ppx/snarkydef.ml

```diff
@@ -21,7 +21,9 @@ let with_label ~local ~loc ~arg exprs =
     else
       match arg with
       | None ->
-          [%expr Snarky_backendless.Checked_ast.with_label]
+          failwith
```

Comment:
Nit: probably worth removing `snarkydef_` and using `snarkydef` everywhere, with the unqualified behaviour as the `snarkydef_` behaviour.